update-versions.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/sh
+
+BASE_DIR=$(cd "$(dirname "$0")" || exit; pwd -P)
+FILE_LIST="/tmp/update_version_files"
+
+print_usage() {
+    printf 2>&1 "\
+Usage: %s [-a] NEW_VERSION
+
+Update version string in mix.exs and config.exs with NEW_VERSION
+by scanning lines with version: string. This script will try its
+best to retain the indention of the original string.
+
+If -a is given, it will try to determine the version from current
+branch and Git version, suitable for running in Ci environment.
+" "$0"
+}
+
+OPTIND=1
+VERSION=
+AUTO=0
+
+while getopts "a" opt; do
+    case "$opt" in
+        a ) AUTO=1;;
+        * ) print_usage; exit 1;;
+    esac
+done
+
+shift $((OPTIND-1))
+if [ "${1:-}" = "--" ]; then
+    shift
+fi
+
+VERSION=$1
+
+if [ "$AUTO" = 1 ]; then
+    if ! command -v git >/dev/null; then
+        printf 2>&1 "Git is required to auto-generating version\\n"
```

Comment:
Sorry this should be `printf >&2` not `2>&1`. I was probably drunk while writing this 😅

Reply:
Applies to the rest as well.